### PR TITLE
Fix deserialization error in verifier

### PR DESCRIPTION
**Pull request by NikhilCollooru**
Stats class doesn't have a constructor that helps deserializing. So instead of 
deserializing into List<Stats> objects, deserialize them into a 
List<Map<String,Object>>. Because all we want is just the size of list. 
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by caithagoras** [comment]
@NikhilCollooru Thanks for the fix. Were you able to tested it locally on your laptop?

**Comment by NikhilCollooru** [comment]
> @NikhilCollooru Thanks for the fix. Were you able to tested it locally on your laptop?

Yes , like you suggested i picked a query and ran it using verifier with this fix. It ran fine.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/NodeResourceClient.java [hunk 1]
```diff
@@ -13,9 +13,7 @@
  */
 package com.facebook.presto.verifier.prestoaction;
 
-import com.facebook.airlift.json.JsonCodec;
-
-public interface PrestoResourceClient
+public interface NodeResourceClient
 {
-    <V> V getJsonResponse(String path, JsonCodec<V> responseCodec);
+    int getSize(String path);
```
Comment (rongrong) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/NodeResourceClient.java]: What does `NodeResourceClient::getSize` mean? As an API this name is very vague.

Comment (NikhilCollooru) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/NodeResourceClient.java]: It's the size of the array of objects in the Json response (list of HeartbeatFailureDetector.Stats objects ).  This size tells us the size of the test cluster which is needed to resolve HIVE_TOO_MANY_OPEN_PARTITIONS error that occurs when the  verifier cluster you're using is smaller than the production cluster.

Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/NodeResourceClient.java]: Maybe we should give it a better name. The method name should be self-explanatory and self-documenting. We can call it `getClusterSize` and `getWorkerCount`.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/HttpNodeResourceClient.java [hunk 2]
```diff
@@ -49,24 +55,33 @@ public HttpPrestoResourceClient(
...
     @Override
-    public <R> R getJsonResponse(String path, JsonCodec<R> responseCodec)
+    public int getSize(String path)
     {
-        return networkRetry.run(format("getJsonResponse()", path), () -> getJsonResponseOnce(path, responseCodec));
+        return networkRetry.run(format("getSize()", path), () -> getSizeOnce(path));
     }
 
-    private <R> R getJsonResponseOnce(String path, JsonCodec<R> responseCodec)
+    private int getSizeOnce(String path)
```
Comment (caithagoras) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/HttpNodeResourceClient.java]: Similar here

File: presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/NodeResourceClient.java [hunk 3]
```diff
@@ -13,9 +13,7 @@
  */
 package com.facebook.presto.verifier.prestoaction;
 
-import com.facebook.airlift.json.JsonCodec;
-
-public interface PrestoResourceClient
+public interface NodeResourceClient
```
Comment (caithagoras) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/NodeResourceClient.java]: Ideally, name changes should be in a separate commit.
